src/crypto/sha256.cpp

```diff
@@ -623,6 +638,35 @@ std::string SHA256AutoDetect()
 #endif
 #endif
 
+#if defined(ENABLE_ARM_SHANI) && !defined(BUILD_BITCOIN_INTERNAL)
+    bool have_arm_shani = false;
+
+#ifdef __linux__
+#ifdef __arm__ // 32-bit
+    if (getauxval(AT_HWCAP2) & HWCAP2_SHA2) {
+        have_arm_shani = true;
+    }
+#endif
+#ifdef __aarch64__ // 64-bit
+    if (getauxval(AT_HWCAP) & HWCAP_SHA2) {
+        have_arm_shani = true;
+    }
+#endif
+#endif
+
+#if defined(__APPLE__)
```

Comment:
a6ae9ef1626c73253aa8e6c8e406f3ee18b8017f

style nit: `#ifdef __linux__` vs `#ifdef __aarch64__` vs `#if defined(__APPLE__)` looks inconsistent. Mind choosing one style for new code?

Reply:
Addressed in 9560981b0dda342b79e25067132dcb37e8d9e27b